src/tests/ftest/server/daos_server_dump.py

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020-2021 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from __future__ import print_function
+
+from apricot import TestWithServers
+from server_utils import ServerFailed
+from general_utils import pcmd, stop_processes
+
+
+class DaosServerDumpTest(TestWithServers):
+    """Daos server dump tests.
+
+    Test Class Description:
+        Simple test to verify that ULT stacks dumps work
+
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DaosServerDumpTest object."""
+        super().__init__(*args, **kwargs)
+        self.start_servers_once = False
+        self.setup_start_agents = False
+        self.setup_start_servers = False
+
+    def test_daos_server_dump_basic(self):
+        """JIRA ID: DAOS-1452.
+
+        Test Description: Test engine ULT stacks dump.
+
+        :avocado: tags=all,hw,large,control,daily_regression,server_start,basic
+        :avocado: tags=test_daos_server_dump_basic
+        """
+        # Setup the servers
+        self.add_server_manager()
+        self.configure_manager(
+            "server", self.server_managers[0], self.hostlist_servers,
+            self.hostfile_servers_slots)
+
+        try:
+            self.server_managers[0].start()
+            exception = None
+        except ServerFailed as err:
+            exception = err
+
+        # Verify
+        if exception is not None:
+            self.log.error("Server was expected to start")
+            self.fail(
+                "Server start failed when it was expected to complete")
+
+        ret_codes = stop_processes(self.hostlist_servers, r"daos_engine",
+                           added_filter=r"'\<(grep|defunct)\>'",
+                           dump_ult_stacks=True)
```

Comment:
In addition to stopping all the engines you'll want to update the expected states of the engines so there are no errors reported in `tearDown()`.  It would make the most sense to add a function to the `DaosServerManager` class, something like this:
```
    def kill_engines(self, daos_log, hosts=None, dump_ult_stacks=False):
        """Kill any engines running on the specified hosts.

        Args:
            daos_log (DaosLog): object for logging messages
            hosts (list, optional): hosts on which to kill the engines. Defaults
                to None which will use all of the hosts.
            dump_ult_stacks (bool, optional): whether to dump ULT stacks.
                Defaults to False.
        """
        if hosts is None:
            hosts = self._hosts
        msg = "Stopping DAOS engines on {} from server group {}".format(
            hosts, self.get_config_value("name"))
        self.log.info(msg)
        daos_log.info(msg)

        # Stop desired ranks using pkill
        stop_processes(
            hosts, r"daos_engine", added_filter=r"'\<(grep|defunct)\>'",
            dump_ult_stacks=dump_ult_stacks)

        # Update the expected status of the stopped/excluded ranks
        self.update_expected_states(
            self.get_host_ranks(hosts), ["stopped", "excluded"])
```
You'll also need to add this method in the same class  https://github.com/daos-stack/daos/blob/429ba4ebb356560581ff786947149959181e6178/src/tests/ftest/util/server_utils.py#L1057-L1071

You could return the return value from the `stop_processes()` call in `DaosServerManager.kill_engines()` but I think running a `self.server_manager[0].verify_expected_states()` after calling `self.server_manager[0].kill_engines(self.d_log, dump_ult_stacks=True)` would be more useful.


See also the comment in src/tests/ftest/util/general_utils.py about possibly moving the sending of the USR2 signal into this method.